### PR TITLE
[fix]: [CCM-14548]: Fix Ec2 Recommendation Batch Job

### PR DESCRIPTION
**Pull request by iamanmolm**
## Describe your changes
Instance Name can be null from EC2 Recommendation fetched from AWS and white setting use UpdateOperations.set we check if it is a null value than we skip the field.
And second if batch job fails for one connector for some issue we were terminating batch job by throwing custom exception, so removing that as well.

## Checklist
- [x] I've documented the changes in the PR description.
- [x] I've tested this change either in PR or local environment.
- [x] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/53643)

### CODE REVIEW

**Comment by iamanmolm** [comment]
trigger uts

**Comment by iamanmolm** [comment]
trigger messagecheck

File: 490-ce-commons/src/main/java/io/harness/ccm/commons/dao/recommendation/EC2RecommendationDAO.java [hunk 1]
```diff
@@ -111,6 +110,10 @@ public EC2Recommendation saveRecommendation(EC2Recommendation ec2Recommendation)
             .set(EC2RecommendationKeys.rightsizingType, ec2Recommendation.getRightsizingType())
             .set(EC2RecommendationKeys.lastUpdatedTime, ec2Recommendation.getLastUpdatedTime());
 
+    if (ec2Recommendation.getInstanceType() != null) {
+      updateOperations.set(EC2RecommendationKeys.instanceType, ec2Recommendation.getInstanceType());
+    }
```
Comment (abhinavsingh-ccm) [inline 490-ce-commons/src/main/java/io/harness/ccm/commons/dao/recommendation/EC2RecommendationDAO.java:115]: The issue is with `instanceName` field 😄. As per our discussion, rather adding the check on DAO layer, please add it to the Tasklet itself. Probably adding blank field should work?

**Review by abhinavsingh-ccm — COMMENTED** [review]
Small comment

**Review by abhinavsingh-ccm — APPROVED** [review]
👍

**Comment by iamanmolm** [comment]
trigger uts

**Pull request merged**